### DISCUSSION TITLE
Why there's no updates, is the project current still?

### DISCUSSION THREAD

**Discussion by rwp0** [RexOps/Rex#1525] [General]
It has been almost a year that there's no release.

And there's no relevant notice on that in the documentation.

Given that Perl 5.36 is released we expect a new release to follow soon.

Let's discuss that here

Thanks

**Comment by ferki** [comment]
Hi @rwp0, thanks for opening a discussion about it!

> Why there's no updates, is the project current still?

Rex is developed by volunteers in their free time, so low amount of available capacity is the most probable reason of perceived low code activity. The project itself stays active, along with the available community support channels.

Speaking for myself, since the last release I have spent ~74 hours on Rex community support and maintenance, and another ~30 hours on Rex extension modules on the CPAN. It's often hard to justify the time spent because that's time taken away from either my family, my own needs, or my own business. I, for one, intend to continue working on and with Rex (or even increase the activity) whenever the investment is justified.

> It has been almost a year that there's no release.

Yeah, I agree that's somewhat weird. I also think it's not necessarily special for purely volunteer-based projects, though. There's not much added to the codebase since last release apart from GitHub (Actions) fixes, so I'm not sure if it's worth to make a release yet.

I do have some patches in the queue that are relevant to my own use cases in my private systems. I look forward to make them available upstream in a generally useful way sometime. Meanwhile, I'm also happy to support contributors to make sure their changes pass all core requirements, or to publish their own extensions.

> And there's no relevant notice on that in the documentation.

Hmm, I'm not sure what kind of notice and on which channel would be useful for this. What would have been useful for you to avoid confusion?

> Given that Perl 5.36 is released we expect a new release to follow soon.

Generally Rex release cadence is not tied to Perl release cadence. Automated tests are running daily on GitHub Action with the latest available Perl version, and I also have daily builds of my maintained projects on my private self-hosted CI system.

If something would be broken by perl-5.36.0, I expect it would have been caught already. Currently all tests are passing on perl-5.36.0, so I don't see anything imminent to do.

Why do you expect a new Rex release for perl-5.36.0? Did the upgrade broke something for you?

**Reply by rwp0** [reply]
Thanks for the detailed response. Before relying on an automation solution I always check if it's current/maintained. Good to hear that there's daily builds and tests in place and esp. that they're [open](https://github.com/RexOps/Rex/actions) for all to view. That's comforting. I think the best notice would be seeing recent commits and releases, another factor for choosing which CPAN modules to use for projects. I will try to get more active with Rex, and involve myself more in discussions, channels, and reports. Thanks again, and good luck with keeping up such a useful and lacking-in-Perl-world project.

**Comment by ferki** [comment]
> Before relying on an automation solution I always check if it's current/maintained.

That sounds like a great practice :+1: For the sake of completeness: as part of my professional services, I also offer (free of charge) open source office hours around my projects, as well as open to provide consulting, training, development, etc. around Rex.

> Good to hear that there's daily builds and tests in place and esp. that they're [open](https://github.com/RexOps/Rex/actions) for all to view. That's comforting.

I'm glad you like it, and thanks for linking! I keep a close eye on the notifications whenever I'm not on a vacation.

> I think the best notice would be seeing recent commits and releases, another factor for choosing which CPAN modules to use for projects.

That's a fair point, thanks. There used to be monthly regular releases, but I do play with the idea of adopting a more rolling model and e.g. release after each passing merge to the default branch. It sometime proved to be too frequent in the past, though, so I guess the right balance lies somewhere ein between (and might move over time).

> I will try to get more active with Rex, and involve myself more in discussions, channels, and reports.

Sounds great, you're welcome to join, and by opening this discussion you are already contributing ;) To get more ideas, the [Contribute to (R)?ex](https://www.rexify.org/care/help__r__ex.html) page and the [Contributing guide](https://github.com/RexOps/Rex/blob/master/CONTRIBUTING.md) might be interesting resources to get started with what's there today.

> Thanks again, and good luck with keeping up such a useful and lacking-in-Perl-world project.

Thanks for your kind words and feedback! All in all, I'm contributing since 9+ years to Rex by now, and I'd like to do so in the next 9+ years too ;)
